proto/lighthouse-result.proto

```diff
@@ -138,10 +149,49 @@ message LighthouseResult {
 
     // List of the categories that were run, empty if all were run
     // nullable list of strings
+    // Note: this should have been google.protobuf.ListValue, but too late now.
     google.protobuf.Value only_categories = 3;
 
     // How Lighthouse was run, e.g. from the Chrome extension or from the npm module
     string channel = 4;
+
+    // The maximum amount of time to wait for a page content render, in ms. If no content is rendered within this limit, the run is aborted with an error.
+    int32 max_wait_for_fcp = 5;
+
+    // The maximum amount of time to wait for a page to load, in ms.
+    int32 max_wait_for_load = 6;
+
+    // List of URL patterns to block.
+    repeated string blocked_url_patterns = 7;
+
+    // Comma-delimited list of trace categories to include.
+    repeated string additional_trace_categories = 8;
+
+    // Flag indicating that the browser storage should not be reset for the audit.
+    bool disable_storage_reset = 9;
+
+    // Flag indicating that there shouldn't be any emulation during the run.
+    bool disable_device_emulation = 10;
+
+    // The method used to throttle the network.
+    ThrottlingMethod throttling_method = 11;
+
+    // The throttling config settings.
+    ThrottlingSettings throttling = 12;
+
+    // If present, the run should only conduct this list of audits.
+    // nullable list of strings
+    google.protobuf.ListValue only_audits = 13;
+
+    // If present, the run should skip this list of audits.
+    // nullable list of strings
+    google.protobuf.ListValue skip_audits = 14;
+
+    // List of extra HTTP Headers to include.
+    map<string, string> extra_headers = 15;
+
+    // Precomputed lantern estimates to use instead of observed analysis.
+    PrecomputedLanternData precomputed_lantern_data = 16;
```

Comment:
I'm not super excited about exposing this in its current form to the world, is everything here going to be available over the PSI API automatically?

Reply:
All these changes will only be exposed in LR atm. But perhaps in the future we'd expand PSI to allow a config. We could disallow whatever settings we want from PSI.

Do you have a refactor of this setting in mind? If so, we should do that before encoding in proto. If this is just a setting that's only useful for testing I don't think the shape matters much.

Reply:
> Do you have a refactor of this setting in mind?

Well in my variance doc I proposed removing it completely depending on what direction we want to head, so I wouldn't want a brand new setting to popup for PSI users only to take it away a few weeks later :)

Reply:
We could drop it completely + disable the smokeTest that uses in Smokerider indefinitely.